learn/building_programs.md

```diff
@@ -0,0 +1,780 @@
+"ldd" or "dependency walker" to find out what libraries are loaded and
+where they are found.
+
+If you want to share your program with colleagues or clients or simply
+users all over the world, you will have to take care that, besides the
+program, you also distribute the libraries it depends on. For more
+information: see below.
+
+
+Include files and modules
+-------------------------
+
+Your program can be contained in various source files, all stored in the
+same directory or organised in some convenient directory tree. The
+details of the organisation depend on personal taste, arrangements made
+by the group of developers you belong to, or simply the history of the
+program. Whatever the directory structure, you will encounter a
+situation where the compiler needs assistence in order to compile a
+particular source file:
+
+* Fortran (and other languages) has the possibility to include an
+external file. While this feature has become a bit less useful with the
+advent of modules, it still has its uses. Quite often, such "include
+files" are stored in a directory separated from the directories
+containing the source files, because they are used in several locations.
+Quite often the name of that directory is "include".
+* As we have seen, source code that defines one or more modules, leads to
+so-called "module intermediate files" (with the extension ".mod"). The
+compiler needs access to these files to be able to read the interfaces
+and variables and what not, and based on this information, actually
+compile the source code that uses the various modules.
+
+Compilers support options like "-I" to indicate where these include
+files and module intermediate files are to be found. Suppose we store
+the two files of our tabulation program in the following directory
+structure:
+
+    tabulate/
+        main/
+            tabulate.f90
+        sub/
+            function.f90
+
+Compiling the file "function.f90" with the commands
+
+    $ cd sub
+    $ gfortran -c function.f90
+
+leads to this structure:
+
+        tabulate/
+            main/
+                tabulate.f90
+            sub/
+                function.f90
+                function.mod
+                function.o
+
+To successfully compile and subsequently build the program we need to
+tell the compiler where it can find the file "function.mod":
+
+    $ cd main
+    $ gfortran -c tabulate.f90 -I ../sub
+    $ gfortran -o tabulate tabulate.o ../sub/function.o
+
+The result:
+
+    tabulate/
+        main/
+            tabulate.f90
+            tabulate.o
+            tabulate (or tabulate.exe on Windows)
+        sub/
+            function.f90
+            function.mod
+            function.o
+
+Notes:
+
+* The details differ per compiler. Sometimes the "-I" option should be
+followed by a space and then the name of the directory, sometimes the
+directory should come consecutively.
+* Many compilers allow you to specify the location for the module

```

Comment:
This is a little too ambiguous for a beginner; can you make the difference between the `-I` flag and the `-J` flag clearer.
_i.e._ that compilers place the .mod files in the current directory by default and that the `-J` flag explicitly specifies an alternative location to place them (and search for them).